src/main/kotlin/com/team4099/robot2023/subsystems/superstructure/Superstructure.kt

```diff
@@ -0,0 +1,990 @@
+              Manipulator.TunableManipulatorStates.groundIntakeCubeExtension.get(),
+              ManipulatorConstants.IDLE_VOLTAGE
+            )
+        }
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition && manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CUBE
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CUBE_CLEANUP
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CUBE -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.intakeAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.intakeVoltage.get()
+          )
+        manipulator.currentRequest =
+          Request.ManipulatorRequest.TargetingPosition(
+            Manipulator.TunableManipulatorStates.groundIntakeCubeExtension.get(),
+            Manipulator.TunableManipulatorStates.cubeInVoltage.get()
+          )
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition && manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CUBE_CLEANUP
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CUBE_CLEANUP
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CUBE_CLEANUP -> {
+        // Goes immediately to IDLE
+        // Outputs
+
+        // Transition
+        nextState = SuperstructureStates.IDLE
+      }
+      SuperstructureStates.GROUND_INTAKE_CONE_PREP -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.stowedUpAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.neutralVoltage.get()
+          )
+        if (groundIntake.isAtTargetedPosition) {
+          manipulator.currentRequest =
+            Request.ManipulatorRequest.TargetingPosition(
+              Manipulator.TunableManipulatorStates.groundIntakeConeExtension.get(),
+              ManipulatorConstants.IDLE_VOLTAGE
+            )
+        }
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition && manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CONE
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CONE_CLEANUP
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CONE -> {
+        // Outputs
+        manipulator.currentRequest =
+          Request.ManipulatorRequest.TargetingPosition(
+            Manipulator.TunableManipulatorStates.groundIntakeConeExtension.get(),
+            Manipulator.TunableManipulatorStates.coneInVoltage.get()
+          )
+
+        // Transition
+        if (manipulator.isAtTargetedPosition) {
```

Comment:
we should not switch to cleanup unless the request changes or we think we have gotten the game piece, right? Otherwise we will immediately transition and never actually intake